### PR TITLE
Dev/{p2 37} meal search page/frontend

### PR DESCRIPTION
**Pull request by dgheig**

### CODE REVIEW

File: app/backend/src/DataFixtures/AbstractDataImportFixture.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+use function file_get_contents;
+use function fopen;
+use function html_entity_decode;
+use InvalidArgumentException;
+use function json_decode;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use function trim;
+
+abstract class AbstractDataImportFixture extends Fixture implements FixtureGroupInterface, LoggerAwareInterface
```
Comment (noahboegli) [inline app/backend/src/DataFixtures/AbstractDataImportFixture.php:22]: Should not include the logger aware

File: app/backend/src/Entity/RecipeIngredient.php [hunk 2]
```diff
@@ -2,63 +2,60 @@
...
-    #[Assert\NotNull]
-    private ?Ingredient $ingredient;
+    #[ORM\Column(type: "float")]
+    private float $quantity;
 
-    #[ORM\ManyToOne(targetEntity: Step::class, inversedBy: "ingredients")]
+    #[ORM\ManyToOne(targetEntity: Recipe::class, inversedBy: "ingredients")]
     #[ORM\JoinColumn(nullable: false)]
-    private ?Step $step;
+    private ?Recipe $recipe;
```
Comment (noahboegli) [inline app/backend/src/Entity/RecipeIngredient.php:17]: Remove `?`

File: app/backend/src/Entity/Unit.php [hunk 3]
```diff
@@ -46,7 +46,8 @@ public static function getUnitTypes(): array
     {
         return [
             "volume",
-            "wight"
+            "weight",
+            "undefined"
```
Comment (noahboegli) [inline app/backend/src/Entity/Unit.php:50]: Remove `undefined`

**Review by noahboegli — CHANGES_REQUESTED** [review]

File: app/backend/src/DataFixtures/AbstractDataImportFixture.php (hunk 1 above)
Comment (nigelmann) [reply app/backend/src/DataFixtures/AbstractDataImportFixture.php:22]: Sorry but the logger is used in sub classes so I will not remove it.

Comment (noahboegli) [reply app/backend/src/DataFixtures/AbstractDataImportFixture.php:22]: The child classes should implement it then.

File: app/backend/src/Entity/RecipeIngredient.php (hunk 2 above)
Comment (nigelmann) [reply app/backend/src/Entity/RecipeIngredient.php:17]: Seems fair

File: app/backend/src/DataFixtures/AbstractDataImportFixture.php (hunk 1 above)
Comment (nigelmann) [reply app/backend/src/DataFixtures/AbstractDataImportFixture.php:22]: I do not agree, please re-review, some changes have been made.

Cheers !

Comment (noahboegli) [reply app/backend/src/DataFixtures/AbstractDataImportFixture.php:22]: Fuck you

File: app/backend/src/DataFixtures/AbstractDataImportFixture.php [hunk 4]
```diff
@@ -36,12 +36,14 @@ public function getCsvData(string $path = "", array $mandatoryFields = [], ?stri
         if (empty($path)) {
             $path = static::CSV_FILE;
             if (empty($path)) {
+                $this->logger->error("Path must not be empty");
```
Comment (noahboegli) [inline app/backend/src/DataFixtures/AbstractDataImportFixture.php:39]: Very nice kanye, I like it a lot!

**Review by noahboegli — APPROVED** [review]

File: app/backend/src/DataFixtures/AbstractDataImportFixture.php (hunk 4 above)
Comment (noahboegli) [reply app/backend/src/DataFixtures/AbstractDataImportFixture.php:39]: #grateful

**Review by Kitsunemma — APPROVED** [review]

**Pull request merged**